### PR TITLE
tests: remove workarounds that add "ubuntu-save" if missing

### PR DESCRIPTION
**Pull request by mvo5**
The gadget snaps now all have a ubuntu-save partition so we don't
need the workarounds anymore that add them in the tests.

### CODE REVIEW

File: tests/lib/nested.sh [hunk 1]
```diff
@@ -518,15 +507,6 @@ nested_create_core_vm() {
                         snap download --basename=pc --channel="20/edge" pc
                         unsquashfs -d pc-gadget pc.snap
                         nested_secboot_sign_gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
-                        if [ "$NESTED_ENABLE_TPM" = "true" ] || [ "${NESTED_ADD_UBUNTU_SAVE:-}" = "true" ]; then
-                            # TODO:UC20: until https://github.com/snapcore/pc-amd64-gadget/pull/51/
-                            # lands there is no ubuntu-save in the gadget, make sure we have one
-                            nested_ensure_ubuntu_save pc-gadget --add
-                            touch ubuntu-save-added
-                        else
-                            nested_ensure_ubuntu_save pc-gadget --remove
```
Comment (bboozzoo) [inline tests/lib/nested.sh:527]: We still want this for uc20-save test that runs without ubuntu-save.

**Comment by bboozzoo** [comment]
```
2020-11-24T07:42:21.2425015Z [   39.002270] snapd[1385]: 2020/11/24 07:28:07.453024 task.go:337: DEBUG: 2020-11-24T07:28:07Z ERROR cannot use gadget: gadget does not support encrypted data: volume "pc" has no structure with system-save role
```
maybe the test was run before relevant gadget revisions were published?

**Comment by mvo5** [comment]
Closing until the gadget is ready

**Comment by anonymouse64** [comment]
The gadget is now ready, this should be reopened

File: tests/lib/nested.sh (hunk 1 above)
Comment (bboozzoo) [reply tests/lib/nested.sh:527]: This breaks tests/nested/manual/core20-save:plain variant. I'll push a fix.

**Review by bboozzoo — APPROVED** [review]
LGTM with the tweaks I pushed

**Pull request merged**